vexim/sitechangesubmit.php

```diff
@@ -1,99 +1,145 @@
+        $success = $sth->execute(array(
+            ':domain_id' => $_POST['allusers'],
+            ':on_avscan' => $_POST['on_avscan']));
+      }
+    }
+    break;
+
+  case 'adminpass':
     if (validate_password($_POST['clear'], $_POST['vclear'])) {
       $query = "UPDATE users SET crypt=:crypt WHERE localpart=:localpart AND domain_id=:domain_id";
       $sth = $dbh->prepare($query);
-      $success = $sth->execute(array(':crypt'=>crypt_password($_POST['clear']),
-            ':localpart'=>$_POST['localpart'], ':domain_id'=>$_POST['domain_id']));
-      if ($success) {
-	header ("Location: site.php?updated={$_POST['domain']}");
-	die;
-      } else {
-	header ("Location: site.php?failupdated={$_POST['domain']}");
-	die;
-      }
+      $success = $sth->execute(array(
+          ':crypt' => crypt_password($_POST['clear']),
+          ':localpart' => $_POST['localpart'],
+          ':domain_id' => $_POST['domain_id']));
     } else {
-      header ("Location: site.php?badpass={$_POST['domain']}");
+      header("Location: site.php?badpass={$_POST['domain']}");
       die;
     }
-  }
+    break;
+
+  default:
+    if (isset($_POST['avscan'])) {
+      $_POST['avscan'] = 1;
+    } else {
+      $_POST['avscan'] = 0;
+      $query = "UPDATE users SET on_avscan=:on_avscan WHERE domain_id=:domain_id";
+      $sth = $dbh->prepare($query);
+      $sth->execute(array(
+            ':domain_id' => $_POST['domain_id'],
+            ':on_avscan' => '0'));
+    }
+
+    if (isset($_POST['spamassassin'])) {
+      $_POST['spamassassin'] = 1;
+    } else {
+      $_POST['spamassassin'] = 0;
+      $query = "UPDATE users SET on_spamassassin=:on_spamassassin WHERE domain_id=:domain_id";
+      $sth = $dbh->prepare($query);
+      $sth->execute(array(
+        ':domain_id' => $_POST['domain_id'],
+        ':on_spamassassin' => '0'));
+    }
 
-// User can specify either UID, or username, the former being preferred.
-// Using posix_getpwuid/posix_getgrgid even when we have an UID is so we
-// are sure the UID exists.
-  if (isset ($_POST['uid'])) {
-    $uid = $_POST['uid'];
-  }
-  if (isset ($_POST['gid'])) {
-    $gid = $_POST['gid'];
-  }
+    if (isset($_POST['enabled'])) {
+      $_POST['enabled'] = 1;
+    } else {
+      $_POST['enabled'] = 0;
+    }
 
-  if ($userinfo = @posix_getpwuid ($uid)) {
-    $uid = $userinfo['uid'];
-  } elseif ($userinfo = @posix_getpwnam ($uid)) {
-    $uid = $userinfo['uid'];
-  } else {
-    header ("Location: site.php?failuidguid={$_POST['domain']}");
-    die;
-  }
+    if (isset($_POST['pipe'])) {
+      $_POST['pipe'] = 1;
+    } else {
+      $_POST['pipe'] = 0;
+    }
 
-  if ($groupinfo = @posix_getgrgid ($gid)) {
-    $gid = $groupinfo['gid'];
-  } elseif ($groupinfo = @posix_getgrnam ($gid)) {
-    $gid = $groupinfo['gid'];
-  } else {
-    header ("Location: site.php?failuidguid={$_POST['domain']}");
-    die;
-  }
+    if ($_POST['max_accounts'] == '') {$_POST['max_accounts'] = '0';}
```

Comment:
Please add an `isset` check and put the assignment on its own line, just like above.